packages/main/src/Toolbar.ts

```diff
@@ -0,0 +1,582 @@
+import litRender from "@ui5/webcomponents-base/dist/renderer/LitRenderer.js";
+import { renderFinished } from "@ui5/webcomponents-base/dist/Render.js";
+import ResizeHandler from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import type { ResizeObserverCallback } from "@ui5/webcomponents-base/dist/delegate/ResizeHandler.js";
+import { getEffectiveAriaLabelText } from "@ui5/webcomponents-base/dist/util/AriaLabelHelper.js";
+import Integer from "@ui5/webcomponents-base/dist/types/Integer.js";
+import "@ui5/webcomponents-icons/dist/overflow.js";
+import type I18nBundle from "@ui5/webcomponents-base/dist/i18nBundle.js";
+import { getI18nBundle } from "@ui5/webcomponents-base/dist/i18nBundle.js";
+
+import {
+	TOOLBAR_OVERFLOW_BUTTON_ARIA_LABEL,
+} from "./generated/i18n/i18n-defaults.js";
+
+import ToolbarTemplate from "./generated/templates/ToolbarTemplate.lit.js";
+import ToolbarCss from "./generated/themes/Toolbar.css.js";
+
+import ToolbarPopoverTemplate from "./generated/templates/ToolbarPopoverTemplate.lit.js";
+import ToolbarPopoverCss from "./generated/themes/ToolbarPopover.css.js";
+
+import ToolbarAlign from "./types/ToolbarAlign.js";
+import ToolbarItemOverflowBehavior from "./types/ToolbarItemOverflowBehavior.js";
+import HasPopup from "./types/HasPopup.js";
+
+import type { ToolbarItem } from "./ToolbarItem.js";
+import { getRegisteredToolbarItem, getRegisteredStyles, getRegisteredStaticAreaStyles } from "./features/ToolbarRegistry.js";
+
+import Button from "./Button.js";
+import Popover from "./Popover.js";
+
+function calculateCSSREMValue(styleSet: CSSStyleDeclaration, propertyName: string): number {
+	return Number(styleSet.getPropertyValue(propertyName).replace("rem", "")) * parseInt(getComputedStyle(document.body).getPropertyValue("font-size"));
+}
+
+function parsePxValue(styleSet: CSSStyleDeclaration, propertyName: string): number {
+	return Number(styleSet.getPropertyValue(propertyName).replace("px", ""));
+}
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ * The <code>ui5-toolbar</code> component is used to create a horizontal layout with items.
+ * The items can be overflowing in a popover, when the space is not enough to show all of them.
+ *
+ * <h3>Keyboard Handling</h3>
+ * The <code>ui5-toolbar</code> provides advanced keyboard handling.
+ * <br>
+ * <ul>
+ * <li>The control is not interactive, but can contain of interactive elements </li>
+ * <li>[TAB] - iterates through elements</li>
+ * </ul>
+ * <br>
+ *
+ * <h3>ES6 Module Import</h3>
+ * <code>import "@ui5/webcomponents/dist/Toolbar";</code>
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webc.main.Toolbar
+ * @extends sap.ui.webc.base.UI5Element
+ * @tagname ui5-toolbar
+ * @public
+ * @since 1.17.0
+ */
+@customElement({
+	tag: "ui5-toolbar",
+	languageAware: true,
+	renderer: litRender,
+	template: ToolbarTemplate,
+	staticAreaTemplate: ToolbarPopoverTemplate,
+	dependencies: [
+		Popover,
```

Comment:
We need a setup, similar to `styles` here - all components must declare their dependencies and the toolbar use them here